atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/metrics/SweepMetricsAssert.java

```diff
@@ -262,11 +264,29 @@ private <T, N> Gauge<N> getGauge(String metricNamespace, String name, Map<String
         MetricName metricName = MetricName.builder()
                 .safeName(metricNamespace + "." + name)
                 .safeTags(tag)
-                .putSafeTags("libraryName", "atlasdb")
-                .putSafeTags("libraryVersion", "unknown")
                 .build();
 
-        return (Gauge<N>) metrics.getTaggedRegistry().getMetrics().get(metricName);
+        return (Gauge<N>) KeyedStream.stream(metrics.getTaggedRegistry().getMetrics())
+                .filterKeys(metric -> areMetricsApproximatelyEqual(metricName, metric))
+                .values()
+                .findFirst()
+                .orElse(null);
+    }
+
+    private boolean areMetricsApproximatelyEqual(MetricName approxMetric, MetricName metric) {
```

Comment:
yeah this is what I had in mind. We might want a more descriptive name than approximately equal since these are not commutative!